applications/UmatApplication/tests/test_J2_plasticity.py

```diff
@@ -0,0 +1,230 @@
+            "constitutive_law": {
+                "law_name"   : "KratosMultiphysics.ConstitutiveModelsApplication.SmallStrain3DLaw",
+                "model_name" : "KratosMultiphysics.UmatApplication.VonMisesSmallStrainUmatModel"
+            },
+            "variables": {
+                "KratosMultiphysics.YOUNG_MODULUS": 10000.0,
+                "KratosMultiphysics.POISSON_RATIO": 0.20,
+                "KratosMultiphysics.YIELD_STRESS": 200.0
+            },
+            "element_type": "Tetrahedra3D4",
+            "nodes" : [ [0.0,0.0,0.0], [1.0,0.0,0.0], [0.0,1.0,0.0], [0.0,0.0,1.0] ],
+            "strain": {
+                "deformation_gradient" : [ [1.0,0.0,0.0], [0.0,1.0,0.0], [0.0,0.0,1.0] ],
+                "jacobian": 1.0
+            },
+            "echo_level" : 0
+
+        }
+        """)
+        self.model = KratosMultiphysics.Model()
+        self.model_part = self.model.CreateModelPart(settings["model_part_name"].GetString())
+        self.echo_level = settings["echo_level"].GetInt()
+
+        #read nodes
+        self.number_of_nodes = settings["nodes"].size()
+        self.nodes = [] #self.model_part.GetNodes()
+        for i in range(0, self.number_of_nodes):
+            node = self.model_part.CreateNewNode(i,settings["nodes"][i][0].GetDouble(),settings["nodes"][i][1].GetDouble(),settings["nodes"][i][2].GetDouble())
+            self.nodes.append(node)
+
+        self.geometry  = KratosMultiphysics.Geometry()
+        self.dimension = 3
+        if( settings["element_type"].GetString() == "Tetrahedra3D4"):
+            if( self.number_of_nodes != 4 ):
+                print(" number of nodes:",self.number_of_nodes," do not matches geometry :", settings["element_type"].GetString() )
+            else:
+                self.geometry = KratosMultiphysics.Tetrahedra3D4(self.nodes[0],self.nodes[1],self.nodes[2],self.nodes[3])
+
+
+        if( settings["element_type"].GetString() == "Triangle2D3"):
+            if( self.number_of_nodes != 4 ):
+                print(" number of nodes:",self.number_of_nodes," do not matches geometry :", settings["element_type"].GetString() )
+            else:
+                self.geometry  = KratosMultiphysics.Triangle2D3(self.nodes[0],self.nodes[1],self.nodes[2])
+                self.dimension = 2
+
+        #material properties
+        self.properties = self.model_part.Properties[settings["properties_id"].GetInt()]
+
+        #read variables
+        self.variables = settings["variables"]
+        for key, value in self.variables.items():
+            variable = self._GetItemFromModule(key)
+            if( value.IsDouble() ):
+                self.properties.SetValue(variable, value.GetDouble())
+            elif( value.IsArray() ):
+                vector_value = KratosMultiphysics.Vector(value.size())
+                for i in range(0, value.size() ):
+                    vector_value[i] = value[i].GetDouble()
+                self.properties.SetValue(variable, vector_value)
```

Comment:
```suggestion
                self.properties.SetValue(variable, value.GetVector())
```

Reply:
I just erased that bit. I copy-pasted it from elsewhere but was not needed here.